src/main/java/me/gaminglounge/guiapi/Pagenation.java

```diff
@@ -3,101 +3,153 @@
         this.inventoryName = inventoryName;
         return this;
     }
-    
+
+    /*
+     * @param items - List of items to be displayed in the pagenation
+     */
     public Pagenation setItems(List<ItemStack> items) {
         this.items = items;
         return this;
     }
 
-    public Pagenation setBack(ItemStack back) {
-        this.back = back;
+    /*
+     * 
+     */
+    public Pagenation setBackInv(Inventory returnInventory) {
+        this.returnInventory = returnInventory;
         return this;
     }
 
-    public Pagenation setNext(ItemStack next) {
-        this.next = next;
-        return this;
-    }
-
-    public Pagenation setClose(ItemStack close) {
-        this.close = close;
-        return this;
-    }
-
-
+    /*
+     * @return the inventory
+     */
     @Override
     public @NotNull Inventory getInventory() {
 
         if (inventoryName == null) {
             this.inventoryName = mm.deserialize("");
         }
 
-        if (placeholderItem == null) {
-            int[] tmp = {0,1,2,3,4,5,6,7,8,9,17,18,26,27,35,36,44,46,47,48,50,51,52};
-            this.placeholderItemIndex = tmp;
-        }
-
         this.inv = Bukkit.createInventory(this, 9 * 6, inventoryName);
 
-        for (int a:placeholderItemIndex) {
-            inv.setItem(a, placeholderItem);
+        int[] tmp = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 17, 18, 26, 27, 35, 36, 44, 46, 47, 48, 50, 51, 52 };
+        for (int a : tmp) {
+            inv.setItem(a,
+                    new ItemBuilder(Material.GRAY_STAINED_GLASS_PANE)
+                            .setName(inventoryName)
+                            .addBothClickEvent("ItemBuilder:cancel")
+                            .build());
         }
 
-        this.inv.setItem(45, back);
+        this.inv.setItem(45,
+                new ItemBuilder(Material.SPECTRAL_ARROW)
+                        .setName(mm.deserialize(Language.getValue(GuiApi.INSTANCE, player, "back")))
+                        .addBothClickEvent("GuiAPI:prevPage")
+                        .build());
 
-        this.inv.setItem(49, close);
+        this.inv.setItem(49,
+                new ItemBuilder(Material.BARRIER)
+                        .setName(mm.deserialize(Language.getValue(GuiApi.INSTANCE, player, "close")))
+                        .addBothClickEvent("GuiAPI:pagenation_return_to_old_inv_or_close")
+                        .build());
 
-        this.inv.setItem(53, next);
+        this.inv.setItem(53,
+                new ItemBuilder(Material.ARROW)
+                        .setName(mm.deserialize(Language.getValue(GuiApi.INSTANCE, player, "next")))
+                        .addBothClickEvent("GuiAPI:nextPage")
+                        .build());
 
         fillPage(0);
 
         return this.inv;
     }
 
+    /*
+     * Changes the displayed items dependend on page
+     * 
+     * @param pageNum - Page number
+     */
     public void fillPage(int pageNum) {
```

Comment:
Javadoc comments should be marked as such.
```suggestion
    /**
     * Changes the displayed items dependend on page
     * 
     * @param pageNum - Page number
     */
    public void fillPage(int pageNum) {
```